src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/UpToDateCheckImplicitConfiguredInput.cs

```diff
@@ -298,11 +298,11 @@ void AddKeys(ImmutableDictionary<string, ImmutableArray<string>> dictionary)
                 copyReferenceInputs = CopyReferenceInputs;
             }
 
-            ImmutableDictionary<string, ImmutableArray<string>> upToDateCheckBuiltItems;
+            ImmutableDictionary<string, ImmutableDictionary<string, ImmutableArray<string>>> upToDateCheckBuiltItemsByKindBySetName;
             ImmutableArray<(string DestinationRelative, string SourceRelative)> copiedOutputFiles;
             if (jointRuleUpdate.ProjectChanges.TryGetValue(UpToDateCheckBuilt.SchemaName, out change) && change.Difference.AnyChanges)
             {
-                var itemsBySet = new Dictionary<string, HashSet<string>>(BuildUpToDateCheck.SetNameComparer);
+                var itemsByKindBySet = new Dictionary<string, Dictionary<string, HashSet<string>>>(BuildUpToDateCheck.SetNameComparer);
```

Comment:
>itemsByKindBySet [](http://example.com/codeflow?start=20&length=16)

Hmm, I read the name itemsByKindBySet as a Cartesian product, meaning the kind would be extracted first then the set, which is the opposite of how it is used.